### PR TITLE
MWPW-157292: CC and DC: Phone number links wrong

### PR DESCRIPTION
**Pull request by suhjainadobe**
The phone number on the following pages is appearing correctly in the text but showing random characters in the link. Previously, using the placeholder for both the display text and link text displayed the number correctly.

Resolves: [[MWPW-157292]](https://jira.corp.adobe.com/browse/MWPW-NUMBER)

Test URLs:

Before: https://main--milo--adobecom.hlx.live/drafts/suhjain/performance/marquee-light
After:https://placeholderOriginNew--milo--suhjainadobe.hlx.live/drafts/suhjain/performance/marquee-light

### CODE REVIEW

File: libs/features/placeholders.js [hunk 1]
```diff
@@ -130,9 +131,15 @@ export async function decoratePlaceholderArea({
   if (!nodes.length) return;
   const config = getConfig();
   await fetchPlaceholders({ placeholderPath, config, placeholderRequest });
-  const replaceNodes = nodes.map(async (textNode) => {
-    textNode.nodeValue = await replaceText(textNode.nodeValue, config);
-    textNode.nodeValue = textNode.nodeValue.replace(/&nbsp;/g, '\u00A0');
+  const replaceNodes = nodes.map(async (nodeEl) => {
+    if (nodeEl.nodeType === Node.TEXT_NODE) {
+      const newValue = await replaceText(nodeEl.nodeValue, config);
+      nodeEl.nodeValue = newValue;
```
Comment (chrischrischris) [inline libs/features/placeholders.js:137]: nit: inline
```suggestion
      nodeEl.nodeValue = await replaceText(nodeEl.nodeValue, config);
```

File: libs/features/placeholders.js [hunk 2]
```diff
@@ -130,9 +131,15 @@ export async function decoratePlaceholderArea({
...
+    } else if (nodeEl.nodeType === Node.ELEMENT_NODE) {
+      const hrefValue = nodeEl.getAttribute('href');
+      const newValue = await replaceText(hrefValue, config);
+      nodeEl.setAttribute('href', newValue);
```
Comment (chrischrischris) [inline libs/features/placeholders.js:141]: super nit
```suggestion
      const hrefVal = await replaceText(nodeEl.getAttribute('href'), config);
      nodeEl.setAttribute('href', hrefVal);
```

**Review by chrischrischris — CHANGES_REQUESTED** [review]
Looks good, needs test case

File: libs/features/placeholders.js (hunk 2 above)
Comment (suhjainadobe) [reply libs/features/placeholders.js]: done

File: libs/features/placeholders.js (hunk 1 above)
Comment (suhjainadobe) [reply libs/features/placeholders.js]: done

**Review by chrischrischris — APPROVED** [review]

**Review by mokimo — APPROVED** [review]

**Pull request merged**